indico/modules/rb_new/controllers/backend/bookings.py

```diff
@@ -153,3 +158,50 @@ def _process(self):
         booking_details['occurrences'] = occurrences
         booking_details['date_range'] = date_range
         return jsonify(booking_details)
+
+
+class RHBookingStateActions(RHRoomBookingBase):
+    def _check_access(self):
+        RHRoomBookingBase._check_access(self)
+        if self.action == 'approve':
+            if not self.booking.can_be_accepted(session.user):
+                raise Forbidden
+        elif self.action == 'reject':
+            if not self.booking.can_be_rejected(session.user):
+                raise Forbidden
+        elif self.action == 'cancel':
+            if not self.booking.can_be_cancelled(session.user):
+                raise Forbidden
+
+    def _process_args(self):
+        self.booking = Reservation.get_one(request.view_args['booking_id'])
+        self.action = request.view_args['action']
+
+    def _process(self):
+        if self.action == 'approve':
+            self.booking.accept(session.user)
+            state = 'approved'
+        elif self.action == 'reject':
+            reason = request.json['reason']
```

Comment:
let's use webargs for this (could be `self.reject()`, with `reject` being decorated with the webargs decorator)